javascript_nodejs/17.proactive-messages/dialogs/mainDialog/index.js

```diff
@@ -90,40 +99,60 @@ class MainDialog {
         }
     }
 
-    async completeJob(storage, turnContext, jobIdNumber) {
+    async completeJob(turnContext, jobIdNumber) {
+        // Get the list of jobs.
+        const jobs = await this.jobsList.get(turnContext, {});
+
         // Read from storage
```

Comment:
Technically, line 104 is getting the state property `jobs[]` from the turn context's state cache, and line 106 is just accessing one of the elements in the retrieved array.